beacon-chain/core/electra/registry_updates.go

```diff
@@ -28,7 +34,37 @@ import (
 //	         for validator in state.validators:
 //	             if is_eligible_for_activation(state, validator):
 //	                 validator.activation_epoch = activation_epoch
-func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state.BeaconState, error) {
-	// TODO: replace with real implementation
-	return state, nil
+func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) error {
+	currentEpoch := time.CurrentEpoch(state)
+	ejectionBal := params.BeaconConfig().EjectionBalance
+	activationEpoch := helpers.ActivationExitEpoch(currentEpoch)
+	vals := state.Validators()
```

Comment:
`state.Validators()` copies the entire validator set and this is very expensive given the current validator set size. 
Have you thought about using `ValidatorsReadOnly` or `ReadFromEveryValidator`

Reply:
Good point. I'll do something about this

Reply:
This is done now. Using the same design as #14130